<pre>
<b>help qhapipf</b>
-------------------------------------------------------------------------------
<p>
<b><u>Title</u></b>
<p>
    <b> Analysis of Quantitative traits using regression and log-linear modelling </b>
<b>&gt; when phase is unknown</b>
<p>
<b><u>Syntax</u></b>
<p>
        <b>qhapipf</b> <i>varlist</i> [if] [using] [<b>,</b> <i>options</i>]
<p>
    <i>options</i>               Description
    -------------------------------------------------------------------------
    Main
      <b>qt</b>(varname)         specifies the dependent variable.
      <b>ipf</b>(<i>string</i>)         specifies the log-linear model for haplotype
                            frequencies.
      <b><u>reg</u></b><b>ress</b>(<i>string</i>)     specifies the regression model for the quantitative
                            trait.
      <b>start</b>               specifies that the starting posterior weights of
                            the EM algorithm are chosen at random.
      <b><u>dis</u></b><b>play</b>             specifies whether to output parameter estimates.
      <b>known</b>               specifies that phase is known.
      <b>phase</b>(varname)      specifies a variable that identifies whether phase
                            is known for a subset of subjects.
      <b>acc</b>(<i>real</i>)           specifies the convergence threshold of the change
                            of the full log-likelihood.
      <b>ipfacc</b>(<i>real</i>)        specifies the convergence theshold of the change in
                            the log-likelihood of the log-linear model.
      <b>nolog</b>               specifies that the likelihood output is supressed.
      <b>model</b>(<i>#</i>)            specifies a label for the log-linear model being
                            fitted.
      <b>lrtest</b>(<i>numlist</i>)     performs a likelihood ratio test between the two
                            models saved by the<b> model()</b> option.
      <b>convars</b>(<i>string</i>)     specifies a list of variables in the constraints
                            file.
      <b>confile</b>(<i>string</i>)     specifies the name of the constraints file.
      <b>mv</b>                  specifies that missing data will be imputed.
      <b>mvdel</b>               specifies that subjects with missing data will be
                            deleted.
      <b>hap</b>(<i>string</i>)         specifies the haplotype of interest.
      <b>menu</b>                specifies that the command is run through a window
                            interface.
    -------------------------------------------------------------------------
<p>
<p>
<b><u>Description</u></b>
<p>
This command models the relationship between a normally distributed continuous
variable in a population-based random sample and individuals' haplotype.  This
command uses an EM algorithm to resolve haplotype phase. Covariates are
constructed from the haplotype and used in a regression model. Additionally the
EM algorithm also handles missing typings assuming MAR.
<p>
There are two distinct models the log-linear model for haplotype frequencies.
Further details of this procedure are found in the stata command <b>hapipf</b>.
Haplotype frequencies are estimated under the assumption of Hardy-Weinberg
Equilibrium.
<p>
The regression model relates the haplotypes to the quantitative trait.  This
model is specified in <b>regress()</b> with the dependent variable specified by the
<b>qt()</b> option.
<p>
The regresssion model takes a syntax to specify the dummy variables for the
regression model. The syntax can specify within-loci, between-loci and
between-chromosome effects.
<p>
<b><u>Latest Version</u></b>
<p>
The latest version is always kept on the SSC website. To install the latest
version click on the following link
<p>
ssc install qhapipf, replace.
<p>
<b><u>Options</u></b>
<p>
<b>ipf</b>(<i>string</i>) specifies the log-linear model for the haplotype frequency model.
It requires special syntax of the form <b>l1*l2+l3</b>. <b>l1*l2</b> allows all the
interactions between the first two loci and locus 3 is independent of them.
This syntax is used in most books on Log-linear modelling, "-" terms and
brackets are not allowed.
<p>
<b><u>reg</u></b><b>ress</b>(<i>string</i>) specifies the regression model.  The program then creates
"dummy" variables for all the effects. A fuller description of this option is
given in the examples.
<p>
<b>start</b> specifies that the starting posterior weights of the EM algorithm are
chosen at random.
<p>
<b><u>dis</u></b><b>play</b> specifies whether to output parameter estimates.
<p>
<b>known</b> specifies that phase is known.
<p>
<b>phase(</b><i>varname</i><b>)</b> specifies a variable that identifies whether phase is known for
a subset of subjects. The variable must contain 1 where phase is known and 0
where phase is unknown.
<p>
<b>acc(</b><i>real</i><b>)</b> specifies the convergence threshold of the change of the full
log-likelihood.
<p>
<b>ipfacc(</b><i>real</i><b>)</b> specifies the convergence theshold of the change in the
log-likelihood of the log-linear model.
<p>
<b>model(</b><i>integer</i><b>)</b> specifies a label for the log-linear model being fitted. This
label is used in the<b> lrtest()</b> option.
<p>
<b>lrtest(</b><i>#,#</i><b>)</b> performs a likelihood ratio test between the two models saved by
the<b> model()</b> option.
<p>
<b>convars(</b><i>string</i><b>)</b> specifies a list of variables in the constraints file.
<p>
<b>confile(</b><i>string</i><b>)</b> specifies the name of the constraints file.
<p>
<b>mv</b> specifies that the algorithm should replace missing locus data (".") with a
copy of each of the possible alleles at this locus. This is performed at the
same stage as the handling of the missing phase when the dataset is expanded
into all possible observations. If this option is not specified but some of the
alleles do contain missing data the algorithm sees the symbol "." as another
allele.
<p>
<b>hap(</b><i>string</i><b>)</b> specifies the haplotype of interest. The dummy variables in the
regression are all related to this haplotype. If the user does not slect a
particular haplotype then one is randomly chosen.
<p>
<b>mvdel</b> specifies that all subjects with missing alleles are deleted.
<p>
<b>menu</b> specifies that the command is run through a window interface.
<p>
<b>qt(</b><i>varname</i><b>)</b> specifies the dependent variable in the regression model.
<p>
<b>nolog</b> specifies that the likelihood output is supressed. 
<p>
<p>
<b><u>Examples of Singlepoint Analyses</u></b>
<p>
To execute the menu interface version of this command type
<p>
. qhapipf,menu
<p>
For the examples I shall assume there are three loci a, b and c .  The pairs of
alleles are contained in the 6 variables a1, a2, b1, b2, c1 and c2.  Let the
quantitative trait variable be y.
<p>
All the models described here all assume that the saturated model is fitted for
the haplotype frequencies. For a single locus <b>a</b> this saturated model is
specified by the option <b>ipf(l1)</b>. Given this the regression models are specified
in the <b>regress()</b> option and the more common models are described below. All the
regression models assume that there are two alleles per locus, multiple alleles
are recoded by the algorithm in terms of an allele of interest and all the rest
are the reference group.
<p>
The one parameter constant model is specified by<b> reg(1)</b>.  To add an additional
parameter that is the additive effect of the allele of interest the model is
specified by the option<b> reg([l1+l1])</b>, where<b> l1</b> represents the first locus in
the<b> varlist</b>.  This is the one-locus single-point additive model (one-locus
SAM).The terms between the [] brackets represent the within locus model, in the
SAM the two chromosomes are independent but have the same parameter for the
allele of interest effect. If the allelic effect depended on the chromosome
then there would be two parameters and this is specified by the option
<b>reg([l1a+l1b])</b>, this is the effect of parental imprinting is not additive.
Additionally the within-locus between-chromosome interaction can be included by
replacing the <b>+</b> symbol with <b>*</b>. This parameter is usually called the dominance
parameter.  The two models become<b> reg([l1*l1])</b> and<b> reg([l1a*l1b])</b>,respectively.
<p>
The commands to fit these models are given below.
<p>
. qhapipf a1 a2, ipf(l1) reg(1) qt(y)
. qhapipf a1 a2, ipf(l1) reg([l1+l1]) qt(y)
. qhapipf a1 a2, ipf(l1) reg([l1a+l1b]) qt(y)
. qhapipf a1 a2, ipf(l1) reg([l1*l1]) qt(y)
. qhapipf a1 a2, ipf(l1) reg([l1a*l1b]) qt(y)
<p>
To test whether locus a is associated with the quantitative trait compare the
two regression models <b>1</b> and <b>[l1+l1]</b>
<p>
. qhapipf a1 a2, ipf(l1) reg([l1+l1]) model(0) qt(y)
. qhapipf a1 a2, ipf(l1) reg(1) model(1) lrtest(0,1) qt(y)
<p>
<b><u>Examples of Multipoint Analyses</u></b>
<p>
When modelling more than one locus there are additionally between-loci
interaction terms.  The within-loci interactions are specified within the []
brackets and the between-loci interactions are specified between the []
brackets.  The two-locus SAM now becomes the model<b> [l1+l1]+[l2+l2]</b>, where the
two loci are independent specified by the ``+'' symbol between the two sets of
brackets.  An extension of this model would allow one between-loci interaction
(or ``haplotype'' effect), this is the two-locus multipoint additive model
(two-locus MAM), this model is specified by the option<b> reg([l1+l1]x[l2+l2])</b>.
Note that the<b> x</b> symbol purely says that there is a between loci interaction and
that the "haplotype" effect is additive. This would be a 4 parameter regression
model: the constant term, the first locus additive effect, the second locus
additive effect and an additive haplotype effect. There is one other between
chromosome "haplotype" effect which is when the "haplotype" can be formed
between chromosomes. This model would be specified by the option
<b>reg([l1+l1]*[l2+l2])</b> and now the "haplotype" effect would not be additive.
<p>
The saturated model that ignores parental imprinting is specified by the option
<b>reg([l1*l1]*[l2*l2])</b>. This model contains between-chromosome interactions.
Between-chromosome interactions can be further divided into within-loci
between-chromosome interactions (dominance parameters) and all between-loci
between-chromsome interactions. The full saturated model including parental
imprinting is specified by the option<b> reg([l1a*l1b]*[l2a*l2b])</b>.
<p>
The commands to fit these models are given below
<p>
2-point SAM
. qhapipf a1 a2 b1 b2, ipf(l1*l2) reg([l1+l1]+[l2+l2]) qt(y)
2-point MAM
. qhapipf a1 a2 b1 b2, ipf(l1*l2) reg([l1+l1]x[l2+l2]) qt(y)
2-point MAM (non-additive haplotype effect)
. qhapipf a1 a2 b1 b2, ipf(l1*l2) reg([l1+l1]*[l2+l2]) qt(y)
2-point saturated model
. qhapipf a1 a2 b1 b2, ipf(l1*l2) reg([l1*l1]*[l2*l2]) qt(y)
2-point saturated model with parental imprinting
. qhapipf a1 a2 b1 b2, ipf(l1*l2) reg([l1a*l1b]*[l2a*l2b]) qt(y)
<p>
The algorithm calculates the haplotype frequencies internally and the
log-linear model option<b> ipf()</b> specifies this model. Generally it is taken to be
the saturated model. It may be advantageous to use an intermediate model to
reduce the number of parameters in the full joint likelihood. This can also be
tested using this command using the likelihood ratio test.
<p>
<b><u>Author</u></b>
<p>
Adrian Mander, MRC Human Nutrition Research, Cambridge, UK.
<p>
Email adrian.mander@mrc-hnr.cam.ac.uk
<p>
<b><u>Also see</u></b>
<p>
On-line: help for hapipf (if installed), ipf (if installed). 
</pre>