<pre>
<b> help clump</b>
-------------------------------------------------------------------------------
<p>
<b><u>Title</u></b>
<p>
<b> Genetic association for chromosome data using a permutation test</b>
<p>
<b><u>Syntax</u></b>
<p>
        <b>clump</b> [<i>varlist</i>] [<b>,</b> <i>options</i>]
<p>
    <i>options</i>               Description
    -------------------------------------------------------------------------
    Main
      <b>lrchi2</b>               specifies whether to use the likelihood ratio
                            test.
      <b><u>noi</u></b><b>se</b>                specifies .
      <b>maxiter</b>              specifies the maximum number of iterations.
    -------------------------------------------------------------------------
<p>
    Note: the first variable is the case/control variable that has to be
    binary. The next pair of variables are the pairs of alleles at the marker
    in question. A line of data represents one subject.
<p>
<b><u>Description</u></b>
<p>
    This program assesses the significance of the departure of observed
    values in a contingency table from the expected values conditional on the
    marginal totals. The present implementation works on <b>2 x N</b> tables and was
    designed for use in genetic case-control association studies, but the
    program should be useful for any<b> 2 x N</b> contingency table, especially
    where<b> N</b> is large and the table is sparse. The significance is assessed
    using a Monte Carlo approach, by performing repeated simulations to
    generate tables having the same marginal totals as the one under
    consideration, and counting the number of times that a chi-squared value
    associated with the real table is achieved by the randomly simulated
    data. This means that the significance levels assigned should be unbiased
    (with accuracy dependent on the number of simulations performed) and that
    no special account needs to be taken of continuity corrections or small
    expected values.
<p>
    The method is described in full in:  Sham PC &amp; Curtis D. 1995. Monte
    Carlo tests for associations between disease and alleles at highly
    polymorphic loci.<i>  Ann Hum Genet.</i>  <b>59</b>: 97-105.
<p>
<b><u>Options</u></b>
        +------+
    ----+ Main +-------------------------------------------------------------
<p>
    <b>lrchi2</b> specifies use of the likelihood ratio test on the 2 by 2 table.
        If a table contains a 0 cell the test statistic is missing.
<p>
    <b><u>noi</u></b><b>se</b> displays chi-squared tables on original data
<p>
    <b>maxiter(</b><i>#</i><b>)</b> max iterations for the permutation test
<p>
<b><u>Examples</u></b>
<p>
    . clump cc a1628 b1628, noi maxiter(200) lrchi2
<p>
<b><u>Author</u></b>
    Adrian Mander, MRC Biostatistics Unit, Cambridge, UK.
    Email adrian.mander@mrc-bsu.cam.ac.uk
<p>
<b><u>Also see</u></b>
<p>
    Related commands
<p>
    HELP FILES              SSC installation links    Description
<p>
    gipf (if installed)     (ssc install gipf)        Graphical
    representation of a log-linear model
    hapipf (if installed)   (ssc install hapipf)      Haplotype frequency
    estimation using log-linear models
</pre>