<pre>
-------------------------------------------------------------------------------
help for <b>multproc</b> and <b>smileplot</b>                                  (Roger Newson)
-------------------------------------------------------------------------------
<p>
<p>
<b><u>Multiple test procedures and smile plots</u></b>
<p>
        <b>multproc</b> [<i>if</i>] [<i>in</i>] [ ,
               <b><u>pu</u></b><b>ncor(</b> { <i>#</i> | <i>scalarname</i> | <i>varname</i> } <b>)</b>
               <b><u>pc</u></b><b>or(</b> { <i>#</i> | <i>scalarname</i> | <i>varname</i> } <b>)</b>
               <b><u>me</u></b><b>thod(</b><i>method_name</i><b>)</b> <b><u>pv</u></b><b>alue(</b><i>varname</i><b>)</b> <b><u>ra</u></b><b>nk(</b><i>newvarname</i><b>)</b>
               <b><u>gpu</u></b><b>ncor(</b><i>newvarname</i><b>)</b> <b><u>cr</u></b><b>itical(</b><i>newvarname</i><b>)</b> <b><u>gpc</u></b><b>or(</b><i>newvarname</i><b>)</b>
               <b><u>nh</u></b><b>cred(</b><i>newvarname</i><b>)</b> <b><u>rej</u></b><b>ect(</b><i>newvarname</i><b>)</b> <b>float</b> <b>fast</b> ]
<p>
        <b>smileplot</b> [<i>if</i>] [<i>in</i>] [ ,
               <b><u>es</u></b><b>timate(</b><i>varname</i><b>)</b> <b><u>logb</u></b><b>ase(</b><i>#</i><b>)</b> <b>maxylabs(</b><i>#</i><b>)</b> <b>xlog</b> <b><u>nl</u></b><b>ine(</b><i>#</i><b>)</b>
               <b><u>pts</u></b><b>ymbol(</b><i>symbol</i><b>)</b> <b><u>ptl</u></b><b>abel(</b><i>varname</i><b>)</b> <b>scatteropts(</b><i>scatter_options</i><b>)</b>
               <b>refopts(</b><i>added_line_options_1</i><b>)</b> <b>nrefopts(</b><i>added_line_options_2</i><b>)</b>
               <b>urefopts(</b><i>added_line_options_3</i><b>)</b> <b>crefopts(</b><i>added_line_options_4</i><b>)</b>
               <b>addplot(</b><i>plot</i><b>)</b> <b>plot(</b><i>plot</i><b>)</b> <b>by(</b><i>varlist</i><b>)</b> <i>multproc_options</i>
               <i>twoway_options</i> ]
<p>
    where <i>multproc_options</i> is a list of options recognised by <b>multproc</b>, and
    <i>twoway_options</i> is a list of options recognised by <b>twoway</b> (see help for
    <i>twoway_options</i>).
<p>
    <b>by</b> <i>varlist</i><b>:</b> can be used with <b>multproc</b> and <b>smileplot</b>.  (See help for <b>by</b>.)
    If <b>by</b> <i>varlist</i><b>:</b> is used, then the log output, and all generated variables,
    are calculated using the specified multiple test procedure within each
    by-group defined by the variables in the <i>varlist</i>.
<p>
<p>
<b><u>Description</u></b>
<p>
    <b>multproc</b> takes, as input, a dataset with one value for each of a set of
    multiple statistical tests of multiple null hypotheses, including a
    variable containing <i>P</i>-values for these tests, and an uncorrected overall
    critical <i>P</i>-value specified by the user, and carries out a multiple test
    procedure.  A multiple test procedure calculates a corrected overall
    critical <i>P</i>-value, which has the feature that an individual null
    hypothesis is considered to be acceptable if and only if its
    corresponding <i>P</i>-value is greater than the corrected overall critical
    <i>P</i>-value.  <b>smileplot</b> takes, as input, a dataset with one observation for
    each of a set of estimated parameters, and data on their estimates and
    <i>P</i>-values.  <b>smileplot</b> calls <b>multproc</b> to carry out a multiple test
    procedure, and then creates a smile plot, with data points corresponding
    to estimated parameters, the corresponding <i>P</i>-values (on a reverse log
    scale) on the <i>Y</i>-axis, and another variable (usually the corresponding
    parameter estimates) on the <i>X</i>-axis.  There are <i>Y</i>-axis reference lines
    corresponding to the uncorrected and corrected overall critical <i>P</i>-values.
    The <i>Y</i>-axis reference line corresponding to the corrected overall critical
    <i>P</i>-value is known as the parapet line.  Data points on or above the
    parapet line correspond to rejected null hypotheses.  There may be a
    reference line on the <i>X</i>-axis corresponding to the value of the parameter
    under a null hypothesis (defaulting to 1 if the <i>X</i>-axis is logged, 0
    otherwise).  The user can therefore see, at a glance, both the
    statistical significance and the practical significance of each parameter
    estimate, and can also see the parapet line as an "upper confidence
    bound" on the <i>Y</i>-axis for how many of the corresponding null hypotheses
    are true.  <b>multproc</b> and <b>smileplot</b> are usually used on datasets with one
    observation per parameter estimate and data on estimates and their
    <i>P</i>-values.  Such datasets may be created (directly or indirectly) by 
    <b>postfile</b>, <b>statsby</b>, <b>parmby</b> or <b>parmest</b>.
<p>
    <b>smileplot</b> changed markedly in style in the transition from Stata Version
    7 to Stata Version 8.  Users who prefer to use the Stata 7 version can
    still do so by using <b>smileplot7</b>.
<p>
<p>
<b><u>Options for </u></b><b><u>multproc</u></b><b><u> and </u></b><b><u>smileplot</u></b>
<p>
    <b>puncor(</b> { <i>#</i> | <i>scalarname</i> | <i>varname</i> } <b>)</b> specifies the uncorrected overall
        critical <i>P</i>-value for statistical significance.  This option may be
        specified either as a number, or as a scalar, or as a variable (in
        which case the variable is expected to contain only one non-missing
        value in the sample or by-group).  If absent, this option is set to
        1-<b>$S_level</b>/100, where <b>$S_level</b> is the value of the currently set
        default confidence level.
<p>
    <b>pcor(</b> { <i>#</i> | <i>scalarname</i> | <i>varname</i> } <b>)</b> specifies the corrected overall
        critical <i>P</i>-value for statistical significance.  This option may be
        specified either as a number, or as a scalar, or as a variable (in
        which case the variable is expected to contain only one non-missing
        value in the sample or by-group).  If absent, this option is set by
        the method specified in the <b>method</b> option (see below).
<p>
    <b>method(</b><i>method_name</i><b>)</b> specifies the multiple test procedure method to be
        used for deriving the corrected <i>P</i>-value threshold from the
        uncorrected <i>P</i>-value threshold.  This option is ignored, and set to
        <b>userspecified</b>, if the <b>pcor</b> option is specified and in the range 0 &lt;=
        <b>pcor</b> &lt;= 1.  Otherwise, if <b>method</b> is absent, then it is set to
        <b>bonferroni</b>.
<p>
    <b>pvalue(</b><i>varname</i><b>)</b> is the name of the variable containing the <i>P</i>-values.  If
        this option is absent, then <b>multproc</b> looks for a variable named <b>p</b> (as
        created by <b>parmby</b> or <b>parmest</b>).  <b>multproc</b> carries out a multiple test
        procedure on all observations selected by the <b>if</b> and/or <b>in</b> qualifiers
        which also have non-missing values for the variable containing the
        <i>P</i>-values.
<p>
    <b>rank(</b><i>newvarname</i><b>)</b> is the name of a new variable to be generated,
        containing, in each observation, the rank of the corresponding
        <i>P</i>-value, from the lowest to the highest.  Tied <i>P</i>-values are ranked
        according to their position in the input dataset.  If <b>by</b> <i>varlist</i><b>:</b> is
        specified with <b>multproc</b>, then the ranks are defined within the
        by-group.
<p>
    <b>gpuncor(</b><i>newvarname</i><b>)</b> is the name of a new variable to be generated,
        containing, in each observation, the uncorrected overall critical
        <i>P</i>-value, as specified by the <b>puncor</b> option, or by the standard
        default if the <b>puncor</b> option is not specified.  This new variable
        will have the same value for all observations in the sample of
        observations used by <b>multproc</b> or <b>smileplot</b>.
<p>
    <b>critical(</b><i>newvarname</i><b>)</b> is the name of a new variable to be generated,
        containing, in each observation, an individual critical <i>P</i>-value
        corresponding to the original <i>P</i>-value in the variable specified by
        <b>pvalue</b>.  The values of the individual critical <i>P</i>-values are defined
        by a non-decreasing function (specified by the <b>method</b> option) of the
        ranks of the corresponding original <i>P</i>-values (generated by the <b>rank</b>
        option).  The corrected overall critical <i>P</i>-value is selected from the
        individual critical <i>P</i>-values in a way specified by the <b>method</b> option,
        depending on whether the method specified is a one-step method, a
        step-down method, or a step-up method.
<p>
    <b>gpcor(</b><i>newvarname</i><b>)</b> is the name of a new variable to be generated,
        containing, in each observation, the corrected overall critical
        <i>P</i>-value, as specified by the <b>pcor</b> option, or by the <b>method</b> option if
        the <b>pcor</b> option is not specified.  If <b>by</b> <i>varlist</i><b>:</b> is specified with
        <b>multproc</b>, then the value of this new variable will be the same in all
        observations within each by-group, but may be different for
        observations in different by-groups, if a step-up or step-down
        procedure is specified by the <b>method</b> option.
<p>
    <b>nhcred(</b><i>newvarname</i><b>)</b> is the name of a new variable to be generated,
        containing, for each observation, an indicator of the credibility of
        the corresponding null hypothesis under the method specified by the
        <b>method</b> option.  This indicator is 1 if the null hypothesis is
        credible, and 0 otherwise.  A null hypothesis is said to be credible
        if its <i>P</i>-value is greater than the corrected overall critical
        <i>P</i>-value.  The set of observations with a value of 1 corresponds to a
        set of credible null hypotheses.  The exact interpretation of the set
        of credible null hypotheses depends on whether the method specified
        controls the family-wise error rate (FWER) or the false discovery
        rate (FDR).
<p>
    <b>reject(</b><i>newvarname</i><b>)</b> is the name of a new variable to be generated,
        containing, for each observation, an indicator of the rejection of
        the corresponding null hypothesis under the method specified by the
        <b>method</b> option.  This indicator is 1 if the null hypothesis is
        rejected, and 0 otherwise.  The new variable generated by the <b>reject</b>
        option is therefore the negation of the new variable generated by the
        <b>nhcred</b> option.
<p>
    <b>float</b> specifies that the individual critical <i>P</i>-value variable specified
        by <b>critical()</b> (if requested) will be created as a <b>float</b> variable.  If
        <b>float</b> is absent, then the <b>critical()</b> variable is created as a <b>double</b>
        variable.  Whether or not <b>float</b> is specified, all generated variables
        are stored to the lowest precision possible without loss of
        information.
<p>
    <b>fast</b> is an option for programmers.  It specifies that <b>multproc</b> and
        <b>smileplot</b> will not take any action so that it can restore the
        original data if the user presses <b>Break</b>.
<p>
<p>
<b><u>Options available for </u></b><b><u>smileplot</u></b><b><u> only</u></b>
<p>
    <b>estimate(</b><i>varname</i><b>)</b> is the name of the variable to be plotted on the
        <i>X</i>-axis, usually containing the parameter estimates corresponding to
        the <i>P</i>-values specified by the <b>pvalue</b> option.  If this option is
        absent, then <b>smileplot</b> looks for a variable named <b>estimate</b> (as
        created by <b>parmby</b> or <b>parmest</b>).  <b>smileplot</b> carries out a multiple test
        procedure by calling <b>multproc</b> for observations with non-missing
        values for the variables specified by the <b>estimate</b> and <b>pvalue</b>
        options, using the <b>if</b> and/or <b>in</b> qualifiers if these are supplied by
        the user.  Note that the variable specified by <b>estimate</b> may contain
        values that are not parameter estimates.  For instance, the
        observations may correspond to genes in a genome scan, the <i>P</i>-values
        may be derived from tests for associations of those genes with a
        disease, and the <i>X</i>-axis variable specified by the <b>estimate</b> option may
        contain the positions of those genes on a chromosome map.
<p>
    <b>logbase(</b><i>#</i><b>)</b> specifies a log base used to define the <i>Y</i>-axis labels.  This
        log base is a factor by which each <i>Y</i>-axis label is divided to arrive
        at the next <i>Y</i>-axis label, where the <i>Y</i>-axis labels are ordered from
        the highest <i>P</i>-value to the lowest <i>P</i>-value.  If absent, this option is
        set to 10, so the Y-labels are set to non-positive powers of 10.  If
        this rule defines too many <i>Y</i>-axis labels, then the <i>Y</i>-axis labels are
        set to be every <i>k</i>th member of the logarithmic series, where <i>k</i> is the
        minimum positive integer such that the number of <i>Y</i>-axis labels
        defined in this way is not too large.
<p>
    <b>maxylabs(</b><i>#</i><b>)</b> specifies the maximum number of <i>Y</i>-axis labels allowed.  If
        this option is not specified, then it is set to 25, so as to be
        similar to the Stata 7 version of <b>smileplot</b>, which the user can still
        use by using <b>smileplot7</b>.  <b>maxylabs</b> is used with <b>logbase</b> to decide the
        default sequence of labels on the left <i>Y</i>-axis.  These are chosen to
        be spaced exponentially, separated by a factor equal to the smallest
        possible power of <b>logbase</b> such that the number of labels is no more
        than <b>maxylabs</b>.  This is usually a sensible default, but it can be
        overridden by the <i>twoway_options</i>.
<p>
    <b>xlog</b> specifies that the <i>X</i>-axis must have a log scale.  It is typically
        used if the parameters estimated are odds ratios or geometric mean
        ratios.  It affects the default value of the <b>nline</b> option (see
        below).  It may be overridden by specifications in an xscale option
        in the <i>twoway_options</i> (see help for <i>axis_scale_options</i> or 
        <i>twoway_options</i>).
<p>
    <b>nline(</b><i>#</i><b>)</b> specifies the position, on the <i>X</i>-axis, of the reference line
        indicating the value of the estimated parameters under the null
        hypothesis.  If <b>nline</b> is unspecified, then it is set to 1 if <b>xlog</b> is
        specified and to 0 otherwise.  This option allows the user to plot
        odds ratios and geometric mean ratios on a linear scale, instead of
        on the more usual log scale.  If <b>nline</b> is set to a missing value by
        specifying <b>nline(.)</b>, then the null reference line is suppressed.
        This is useful for creating "smile plots" for which the <i>X</i>-axis
        variable specified by the <b>estimate</b> option contains values other than
        parameter estimates, such as positions of genes on a chromosome map.
<p>
    <b>ptsymbol(</b><i>symbol</i><b>)</b> specifies a graph marker symbol for the data points of
        the smile plot.  This may be any one of the marker symbols named in
        the help for <i>symbolstyle</i>.  If absent, it is set to <b>Th</b> (hollow
        triangles).  This option may be overridden by the option <b>scatteropts</b>
        (see below).
<p>
    <b>ptlabel(</b><i>varname</i><b>)</b> specifies a variable to be used to label the data points
        in the manner specified by the <b>mlabel</b> graph option (see help for 
        <i>marker_label_options</i>).  If this option is absent, then there are no
        data point labels, only unlabelled data points.  This option may be
        overridden by the option <b>scatteropts</b> (see below).
<p>
    <b>scatteropts(</b><i>scatter_options</i><b>)</b> specifies a sequence of options for the
        twoway scatter plot type.  These options may include <b>msymbol</b> and
        <b>mlabel</b> options, which override the <b>ptsymbol</b> and <b>ptlabel</b> options,
        respectively, and other options specifying non-default attributes for
        the symbols and/or labels, such as size and color.  (See help for 
        <b>scatter</b>).  The user may specify any of these options except for <b>xaxis</b>
        or <b>yaxis</b>, because <b>smileplot</b> automatically sets the first <i>X</i>-axis to be
        the $X$-axis of the smile plot (specified by the <b>estimate</b> option) and
        the first and second <i>Y</i>-axes to be the left and right <i>Y</i>-axes used by
        the smile plot (corresponding to the <b>pvalue</b> option).  The second
        <i>Y</i>-axis is used to display the values of the uncorrected and corrected
        overall critical <i>P</i>-values.
<p>
    <b>refopts(</b><i>added_line_options_1</i><b>)</b> specifies a list of added line suboptions,
        as allowed for the <b>xline</b> or <b>yline</b> options.  These options control the
        style of the <i>X</i>-axis and <i>Y</i>-axis reference lines of the smile plot,
        corresponding to the null hypothesis, the uncorrected overall
        critical <i>P</i>-value, and the corrected overall critical <i>P</i>-value,
        respectively.  The options specified by <b>refopts</b> apply to all 3 of
        these reference lines, except if overridden by the <b>nrefopts</b>, <b>urefopts</b>
        and/or <b>crefopts</b> options (see below).  If <b>refopts</b> is absent, then it
        is set to a default, namely <b>refopts(lstyle(xyline))</b>.  This causes the
        lines to have a default style depending on the scheme.  (See help for
        <b>schemes</b> and <b>linestyle</b>.)
<p>
    <b>nrefopts(</b><i>added_line_options_2</i><b>)</b> specifies a list of added line suboptions,
        as allowed for the <b>xline</b> or <b>yline</b> options.  These options control the
        style of the <i>X</i>-axis reference line of the smile plot, corresponding
        to the null hypothesis.  The options specified by <b>nrefopts</b> override
        those specified by <b>refopts</b>, allowing the null hypothesis reference
        line to have a different style from the other reference lines.
<p>
    <b>urefopts(</b><i>added_line_options_3</i><b>)</b> specifies a list of added line suboptions,
        as allowed for the <b>xline</b> or <b>yline</b> options.  These options control the
        style of the <i>Y</i>-axis reference line of the smile plot indicating the
        uncorrected overall critical <i>P</i>-value.  The options specified by
        <b>urefopts</b> override those specified by <b>refopts</b>, allowing the
        uncorrected <i>P</i>-value reference line to have a different style from the
        other reference lines.
<p>
    <b>crefopts(</b><i>added_line_options_4</i><b>)</b> specifies a list of added line suboptions,
        as allowed for the <b>xline</b> or <b>yline</b> options.  These options control the
        style of the <i>Y</i>-axis reference line of the smile plot indicating the
        corrected overall critical <i>P</i>-value.  The options specified by
        <b>crefopts</b> override those specified by <b>refopts</b>, allowing the
        uncorrected <i>P</i>-value reference line to have a different style from the
        other reference lines.
<p>
    <b>addplot(</b><i>plot</i><b>)</b> provides a way to add other plots to the generated graph.
        See help for <i>addplot_option</i>.
<p>
    <b>plot(</b><i>plot</i><b>)</b> is an obsolete alternative name for the <b>addplot()</b> option, used
        by earlier versions of <b>smileplot</b>.  It is provided so that old
        do-files will still run.
<p>
    <b>by(</b><i>varlist</i><b>)</b> is a graph twoway option, and works as for graph twoway,
        creating one subplot for each by-group, arranged in an array as
        specified by the user.  (See help for the by-option.) The corrected
        overall critical <i>P</i>-value, indicated by a line at the same level on
        all the subplots, is calculated from all the <i>P</i>-values from all the
        by-groups pooled together, not for the subset of <i>P</i>-values in each
        by-group individually.  (This is in contrast to the use of <b>by</b>
        <i>varlist</i><b>:</b>, which causes corrected individual and overall critical
        <i>P</i>-values to be calculated only from the subset of <i>P</i>-values in each
        by-group.)
<p>
<p>
<b><u>Remarks</u></b>
<p>
    Multiple test procedures and smile plots are reviewed in Newson <i>et al.</i>
    (2003).  An alternative method, for some multiple-test procedures, is the
    method of frequentist <i>q</i>-values, implemented using the SSC package <b>qqvalue</b>
    and described in Newson (2010).
<p>
    The smile plot is so named because, if the standard errors of the
    parameters are similar, then the data points fall along a curve shaped
    like a smile.  It summarises a set of multiple parameter estimates
    graphically, in the way that a Cochrane forest plot summarises a
    meta-analysis.  The <i>Y</i>-axis reference line corresponding to the corrected
    overall critical <i>P</i>-value is known as the parapet line.  Data points on or
    above the parapet line correspond to parameters for which we can reject
    the null hypotheses under the specified multiple test procedure.  Data
    points below the parapet line correspond to parameters for which the null
    hypotheses are credible (acceptable).
<p>
    The methods specified by the <b>method</b> option are multiple test procedures
    for defining an upper confidence bound for the set of null hypotheses
    that are true, given multiple parameter estimates with multiple <i>P</i>-values.
    More formally, each method defines a set of credible (or acceptable) null
    hypotheses and a set of incredible (rejected) null hypotheses, whose
    exact interpretation depends on the method.  The uncorrected overall
    <i>P</i>-value may either be treated as an upper bound for the family-wise error
    rate (FWER), or be treated as an upper bound for the false discovery rate
    (FDR).
<p>
    The FWER is the probability that at least one true null hypothesis is
    rejected.  If a method controls the FWER, then the power set of the set
    of credible null hypotheses is a power-set-valued confidence region for a
    set-valued parameter, namely the set of null hypotheses which are true.
    We can therefore say, with a confidence level of 100*(1-<b>puncor</b>) percent,
    that the set of null hypotheses that are true is some subset (possibly
    empty) of the set of credible null hypotheses.  In other words, we are
    100*(1-<b>puncor</b>) percent confident that all the rejected null hypotheses
    are false.  FWER-controlling procedures are reviewed in Wright (1992).
<p>
    The FDR is defined as follows.  Let <i>V</i> denote the number of true null
    hypotheses rejected, and let <i>R</i> denote the total number of null hypotheses
    rejected.  Then the FDR is equal to the expectation of <i>Q</i>, where <i>Q</i> is
    defined to be equal to <i>V/R</i> if <i>R&gt;0</i>, and equal to zero if <i>R=0</i>.  The
    probability that <i>Q=1</i> can be no more than the FDR.  Therefore, if the
    method controls the FDR, then we can say, with 100*(1-<b>puncor</b>) percent
    confidence, that the set of null hypotheses that are true is a subset of
    null hypotheses (possibly empty) which does not contain the rejected set
    as a non-empty subset.  In other words, we are 100*(1-<b>puncor</b>) percent
    confident that at least some of the rejected null hypotheses are false.
    If the number of null hypotheses tested is very large indeed, then,
    arguably, we may be 100 percent confident that 100*(1-<b>puncor</b>) percent of
    the rejected null hypotheses are false.
<p>
    The methods may also be classified into one-step, step-down and step-up
    procedures.  All three classes of methods work by defining a list of <i>m</i>
    individual critical <i>P</i>-values <i>C_1,...,C_m</i>, one for each of the <i>m</i>
    individual input <i>P</i>-values <i>P_1,...,P_m</i>, ranked from the lowest to the
    highest.  These individual critical <i>P</i>-values can be saved as output using
    the <b>critical</b> option, and are defined as a non-decreasing function of the
    ranks of the original <i>P</i>-values, which can be saved as output using the
    <b>rank</b> option.  An overall corrected critical <i>P</i>-value <b>pcor</b> is selected from
    the individual critical <i>P</i>-values.  A null hypothesis is acceptable if and
    only if its <i>P</i>-value is greater than the overall corrected critical
    <i>P</i>-value.  For a one-step procedure, the <i>C_i</i> are all equal to the overall
    corrected critical <i>P</i>-value <b>pcor</b>, which is defined as a function of the
    uncorrected critical <i>P</i>-value <b>puncor</b>.  For a step-down procedure, <b>pcor</b> is
    equal to the lowest <i>C_i</i> such that <i>P_i &gt; Ci</i>, if such a <i>C_i</i> exists, and
    equal to <i>C_m</i> otherwise.  For a step-up procedure, <b>pcor</b> is equal to the
    highest <i>C_i</i> such that <i>P_i &lt;= C_i</i>, if such a <i>C_i</i> exists, and equal to <i>C_1</i>
    otherwise.
<p>
    The different methods use different assumptions.  Some assume that the
    different <i>P</i>-values are statistically independent, others allow the
    different <i>P</i>-values to be non-negatively correlated, and others allow the
    different <i>P</i>-values to be arbitrarily correlated.  The more recently
    developed methods are documented in their original source papers.  The
    available methods are as follows:
<p>
<p>
<b>Method</b>          <b>Step type</b>   <b>FWER/FDR</b>    <b>Definition or source</b>
userspecified   One-step    Either      <b>pcor</b> option
bonferroni      One-step    FWER        <b>pcor=puncor/m</b>
sidak           One-step    FWER        <b>pcor=1-(1-puncor)^(1/m)</b>
                                        (or Sidak, 1967)
holm            Step-down   FWER        Holm, 1979
holland         Step-down   FWER        Holland and Copenhaver, 1987
liu1            Step-down   FDR         Benjamini and Liu, 1999a
liu2            Step-down   FDR         Benjamini and Liu, 1999b
hochberg        Step-up     FWER        Hochberg, 1988
rom             Step-up     FWER        Rom, 1990
simes           Step-up     FDR         Benjamini and Hochberg, 1995 (or
                                        Benjamini and Yekutieli, 2001
                                        (first method))
yekutieli       Step-up     FDR         Benjamini and Yekutieli, 2001
                                        (second method)
krieger         Step-up     FDR         Benjamini, Krieger and Yekutieli, 2001
<p>
    Note that, in the case of the <b>holland</b> method, the procedure used is the
    simplified (and less powerful) version of the procedure of Holland and
    Copenhaver (1987), which takes no account of logical dependencies between
    the null hypotheses, although it takes advantage of non-negative
    dependencies between the <i>P</i>-values.  The <b>simes</b> method is so named because
    it was proposed in Simes (1986), although its justification in terms of
    the FDR was presented in the references indicated above.
<p>
<p>
<b><u>Technical note</u></b>
<p>
    If the user specifies <b>method(sidak)</b>, then <b>multproc</b> and {cmd:smileplot)
    use the formula for <b>method(bonferroni)</b> to calculate the corrected
    critical <i>P</i>-values when the Sidak formula gives a corrected critical
    <i>P</i>-value of 0.  Similarly, if the user specifies <b>method(holland)</b>, then
    <b>multproc</b> and {cmd:smileplot) use the formula for <b>method(holm)</b> to
    calculate the corrected critical <i>P</i>-values when the Holland-Copenhaver
    formula gives a corrected critical <i>P</i>-value of 0.  This practice works
    because the Sidak correction converges in ratio to the Bonferroni
    correction in the limit as the uncorrected <i>P</i>-value tends to zero.  This
    approximation makes <b>multproc</b> and <b>smileplot</b> conservative when the
    uncorrected <i>P</i>-value is not very small and the number of comparisons is
    very large.  However, it is less conservative than setting the corrected
    critical <i>P</i>-value to 0.  This technical issue might be a reason for using
    the SSC package <b>qqvalue</b> (Newson, 2010), instead of <b>multproc</b> and
    {cmd:smileplot).
<p>
<p>
<b><u>Examples</u></b>
<p>
    If we type the following example, which uses the <b>auto</b> data, then a smile
    plot will be produced with 1 observation per parameter of the fitted
    model.  The corrected <i>P</i>-value defines an upper confidence bound for how
    many of these parameters are 0 in the population from which these cars
    were sampled.
<p>
                . sysuse auto,clear
                . parmby "xi:regress mpg i.rep78 i.foreign",label norestore
                . smileplot if parm!="_cons",me(holm) ptl(label)
<p>
    If we type the following example, which uses the <b>auto</b> data, then a pair
    of smile plots will be created, one for US-made cars and one for non-US
    cars, with one data point for each parameter of the model (other than the
    intercept).  The corrected <i>P</i>-value is corrected for the total number of
    parameters for both car types (US and non-US).  The pair of smile plots
    is created in two ways, first with the default options, then with a large
    number of non-default options specified by the <b>scatteropts</b> and <b>refopts</b>
    options and the <i>twoway_options</i>.
<p>
                . sysuse auto,clear
                . parmby "xi:regress mpg weight i.rep78",label norestore
                    by(foreign)
                . smileplot if parm!="_cons",ptl(parm) by(foreign)
                . smileplot if parm!="_cons",ptl(parm)
                    scatteropts(mlabsize(medlarge))
                    refopts(lpattern(shortdash)) by(foreign)
                    subtitle(,bfcolor(white)) ylabel(,axis(1)
                    labsize(medlarge)) ylabel(,axis(2) labsize(medlarge))
<p>
    The following advanced example demonstrates the use of <b>by</b> <i>varlist</i><b>:</b>
    together with the <b>by</b> option of <b>smileplot</b>.  The example assumes that there
    is a dataset in memory, with 1 observation per parameter estimate.  The
    dataset contains variables <b>or</b> and <b>siglev</b>, containing estimated odds
    ratios and <i>P</i>-values respectively, and also identifier variables <b>outcome</b>,
    <b>exposure</b>, <b>subset</b> and <b>adjusted</b>.  The program <b>multproc</b> is used to carry out
    the Simes method on each subset defined by the variable <b>adjusted</b>, storing
    the uncorrected and corrected overall critical <i>P</i>-values in new variables
    <b>uncp</b> and <b>corp</b>, and a hypothesis rejection indicator in a new variable
    <b>signif</b>.  We then use <b>smileplot</b> to create, for each combination of values
    of <b>adjusted</b> and <b>outcome</b>, an array of smile plots for each value of
    <b>subset</b>, with data points labelled by the value of <b>exposure</b>.  Finally, the
    rejected null hypotheses are listed.
<p>
                . sort adjusted outcome subset exposure
                . by adjusted:multproc,pval(siglev) meth(simes) gpunc(uncp)
                    gpcor(corp) rej(signif)
                . by adjusted outcome:smileplot,est(or) pval(siglev) punc(uncp)
                    pcor(corp) by(subset) ptl(exposure) xlog t1(" ")
                . by adjusted outcome:list if signif,nodisp
<p>
<p>
<b><u>Saved results</u></b>
<p>
    <b>multproc</b> and <b>smileplot</b> save the following results in <b>r()</b>:
<p>
    Scalars        
      <b>r(puncor)</b>      uncorrected critical <i>P</i>-value
      <b>r(pcor)</b>        corrected critical <i>P</i>-value
      <b>r(npvalues)</b>    number of <i>P</i>-values
      <b>r(nreject)</b>     number of <i>P</i>-values rejected
<p>
    Macros         
      <b>r(method)</b>      the <b>method()</b> option
<p>
<p>
<b><u>Author</u></b>
<p>
    Roger Newson, Imperial College London, UK.  Email:
    r.newson@imperial.ac.uk
<p>
<p>
<b><u>References</u></b> 
<p>
    Benjamini, Y. and Y. Hochberg.  1995.  Controlling the false discovery
        rate: a practical and powerful approach to multiple testing. <i>Journal</i>
        <i>of the Royal Statistical Society B</i> 57: 289-300.
<p>
    Benjamini Y., A. Krieger, and D. Yekutieli.  2001.  Two staged linear
        step-up FDR controlling procedure.  Pre-publication draft
        downloadable from Yoav Benjamini's website at
        http://www.math.tau.ac.il/~ybenja/.
<p>
    Benjamini, Y. and W. Liu.  1999a.  A step-down multiple hypotheses
        testing procedure that controls the false discovery rate under
        independence.  <i>Journal of Statistical Planning and Inference</i> 82:
        163-170.  Pre-publication draft downloadable from Yoav Benjamini's
        website at http://www.math.tau.ac.il/~ybenja/.
<p>
    Benjamini, Y. and W. Liu.  1999b.  A distribution-free multiple-test
        procedure that controls the false discovery rate.  Report, Dept. of
        Statistics and OR, Tel Aviv University, RP-SOR-99-3.  Pre-publication
        draft downloadable from Yoav Benjamini's website at
        http://www.math.tau.ac.il/~ybenja/.
<p>
    Benjamini, Y. and D. Yekutieli.  2001.  The control of the false
        discovery rate in multiple testing under dependency.  <i>Annals of</i>
        <i>Statistics</i> 29: 1165-1188.  Pre-publication draft downloadable from
        Yoav Benjamini's website at http://www.math.tau.ac.il/~ybenja/.
<p>
    Hochberg, Y.  1988.  A sharper Bonferroni procedure for multiple tests of
        significance.  <i>Biometrika</i> 75: 800-802.
<p>
    Holland, B. S. and Copenhaver, M. D.  1987.  An improved sequentially
        rejective Bonferroni test procedure.  <i>Biometrics</i> 43: 417-423.
<p>
    Holm, S.  1979.  A simple sequentially rejective multiple test procedure.
        <i>Scandinavian Journal of Statistics</i> 6: 65-70.
<p>
    Newson, R. B.  2010.  Frequentist <i>q</i>-values for multiple-test procedures.
        <i>The Stata Journal</i> 10(4): 568-584.  Download from <i>The Stata Journal</i>
        website.
<p>
    Newson, R. and the ALSPAC Study Team.  2003.  Multiple-test procedures
        and smile plots.  <i>The Stata Journal</i> 3(2): 109-132.  Download from <i>The</i>
        <i>Stata Journal</i> website.
<p>
    Rom, D. M.  1990.  A sequentially rejective test procedure based on a
        modified Bonferroni inequality.  <i>Biometrika</i> 77: 663-665.
<p>
    Sidak, Z.  1967.  Rectangular confidence regions for the means of
        multivariate normal distributions.  <i>Journal of the American</i>
        <i>Statistical Association</i> 62: 626-633.
<p>
    Simes, R. J.  1986.  An improved Bonferroni procedure for multiple tests
        of significance.  <i>Biometrika</i> 73: 751-754.
<p>
    Wright, S. P.  1992.  Adjusted <i>P</i>-values for simultaneous inference.
        <i>Biometrics</i> 48: 1005-1013.
<p>
<p>
<b><u>Also see</u></b>
<p>
 Manual:  <b>[R] by</b>, <b>[R] statsby</b>, <b>[G] symbol</b>, <b>[G] graph</b>, <b>[G] graph options</b>.
On-line:  help for <b>by</b>, <b>statsby</b>, <b>postfile</b>, <b>graph</b>, <i>graph_twoway</i>
          help for <b>smileplot7</b>, <b>parmby</b>, <b>parmest</b>, <b>qqvalue</b> if installed
</pre>