<pre>
<b>help genefetch</b>                                                   (version 1.2) 
Daniel Elwood Cook {Danielecook@gmail.com}                   www.Elwoodcook.com
-------------------------------------------------------------------------------
<p>
    <b>genefetch</b> --<b> Retreive information from Pub Med for a set of Entrez Gene</b>
    <b>id's or Gene Names. The following information is retrieved:</b>
<p>
         Gene Name
         Gene ID
         Full Name
         Species
         Location
         From (start bp)
         To (end bp)
         Chromosome
<p>
<a name="syntax"></a><b><u>Syntax</u></b>
<p>
    <b>genefetch</b> Gene_Names_OR_IDs <i>, bundle(integer)</i> <i>organism(string)</i>
<p>
<a name="description"></a><b><u>Description</u></b>
<p>
    <b>genefetch</b> uses the efetch utilities provided by the NCBI (National Center
    for Biotechnology Information) to retrieve information for a given list
    of Gene ID's or Gene Names (Single Nucleotide Polymorphims).
<p>
    You should use <b>genefetch</b> for data management/retrieval purposes
    primarily. It creates several variables with set names and fills them
    with data. It will <b>overwrite</b> pre-existing data in these variables
    (example: Gene_ID, Chr_ID).
<p>
    Ideally, you'll use this plugin to annotate a dataset you have by running
    it on only the set of snps from the set and merging or by taking care
    with the names of the variables in your set.
<p>
    <b>genefetch</b> can handle duplicates.
<p>
<a name="usage"></a><b><u>Usage</u></b>
<p>
<b> Method 1: Providing Gene Names</b>
<p>
    There are two ways to use<b> genefetch</b>. One is to provide a list of gene
    names. If you do this,<b> genefetch</b> will retrieve gene id's by performing
    searches on Pubmed's gene database and retrieving the top result.
<p>
    <b>(!Important!)</b> This method has the potential to retrieve incorrect
    information. However, as long as you use the HGNC approved symbol for
    your gene,<b> genefetch</b> should always retrieve the correct record.
<p>
    <b>(!Important!)</b> If you do provide a list of Gene Names and are working with
    a species other than homo sapiens, you<b> must</b> use the <b>organism</b> option,
    specifying the taxonomic name (e.g. Mus musculus, Rattus norvegicus,
    Nomascus leucogenys). If you do not specify an organism, it is set to
    <i>homo sapien</i>.
<p>
    I would recommend using a variable named<b> gene</b> containing a list of Gene
    names when using this method. This variable will remain intact when Gene
    information is retrieved and you can compare the <b>Gene</b> and <b>Gene_Name</b>
    variables to ensure that the correct record has been retrieved.
<p>
<b> Method 2: Providing a list of Gene IDs</b>
<p>
    The second way to use <b>genefetch</b> is to provide it with a list of Entrez
    Gene ID's. This method will always pull in the correct Gene information
    provided the id exists.
<p>
    Follow the command with the name of a variable specifying a list of SNPs.
    Your list can be numeric, ignoring the rs prefix, or a string with or
    without the rs prefix.<b>  genefetch</b> will work with both and download a
    large amount of data for each set.
<p>
    The option<b> bundle</b> can be used to adjust how many records Stata will
    attempt to download at once.<b>  genefetch</b> works by downloading a set of
    records for your list of snps, parsing each set out, add the data to your
    dataset, and move onto the next set. This was necessary due to string
    length limitations within Stata.
<p>
</pre>