<pre>
-------------------------------------------------------------------------------
help for <b>swblock</b>
-------------------------------------------------------------------------------
<p>
<b><u>Stepwise hapipf routine to identify the parsimonious model to describe the Hapl</u></b>
<b><u>&gt; otype block pattern</u></b>
<p>
        <b>swblock</b> [<i>varlist</i>] [, <b>mv(</b><i>string</i><b>)</b> <b><u>p</u></b><b>value(</b><i>#</i><b>)</b> <b><u>s</u></b><b>top</b> <b><u>n</u></b><b>oise</b> <b><u>acc</u></b><b>(</b><i>#</i><b>)</b> <b><u>ipf</u></b><b>acc(</b><i>#</i><b>)</b>
                           <b>store</b> <b>replace</b> ]
<p>
<p>
<b><u>Description</u></b>
<p>
This command systematically fits a series of <b>hapipf</b> log-linear models that
models the LD structure within a set of loci.
<p>
The log-linear model is fitted using iterative proportional fitting which is
available using {hi ssc} and is called <b>ipf</b> (version 1.36 or later).
Additionally, the user will also have to install <b>hapipf</b> (version 1.44 or
later). This algorithm can handle very large contingency tables and converges
to maximum likelihood estimates even when the likelihood is badly behaved.
<p>
If you are connected to the Web you can install the latest version by clicking 
&gt; here 
ssc install hapipf
<p>
The <b>varlist</b> consists of paired variables representing the alleles at each
locus. If phase is known then the paired variables are in fact the genotypes.
When phase is unknown the algorithm assumes Hardy Weinberg Equilibrium so that
models are based on chromosomal data and not genotypic data.
<p>
This algorithm can handle missing alleles at the loci by using the <b>mv()</b> option.
<p>
<b><u>Options</u></b>
<p>
<b>mv(</b><i>string</i><b>)</b> specifies how the missing data will be handled, the default is <b>mv</b>.
If the <i>string</i> is <b>mv</b>, i.e <b>mv(mv)</b>, then the missing data will be assumed to be
missing at random (MAR) and the EM algorithm expands the unknown phase to
consider all possible values for the missing value.  The main assumption of
this algorithm is that the missing data can only take the alleles observed for
a given loci. Relaxing this assumption would not make any difference because
alleles that are never observed usually give expected frequencies that are
close to 0, however, it would increase the number of cells and hence reduce
power. The only other <b>string</b> this option takes is <b>mvdel</b>, i.e <b>mv(mvdel)</b> here the
missing data are assumed to be missing completely at random (MCAR) and subjects
are deleted when they contain any missing data at any loci. Under this
assumption complete subjects are representative of the whole dataset and hence
deletion will give unbiased estimates.
<p>
<b><u>s</u></b><b>top</b> specifies that the search should stop when the inclusion of minimum high
order LD terms do not significantly change the log likelihood.  For example if
none of the third order LD terms included in the model were significant then
the algorithm will not fit the fourth order terms.
<p>
<b><u>acc</u></b><b>(</b><i>#</i><b>)</b> specifies the tolerance of <b>hapipf</b> convergence. The default is 0.0001.
<p>
<b><u>ipf</u></b><b>acc(</b><i>#</i><b>)</b> specifies the tolerance of <b>hapipf</b> convergence. The default is 1.000e-
&gt; 07.
<p>
<b><u>p</u></b><b>value(</b><i>#</i><b>)</b> specifies the significance level for inclusion to the model; terms wi
&gt; th p&gt;pvalue() are not eligible for inclusion.
<p>
<b><u>n</u></b><b>oise</b> specifies that the test statistic values are included in the output
<p>
<b>store</b> specifies that all the model output is saved to a file called <b>fresults.dt</b>
<b>&gt; a</b>
<p>
<b>replace</b> specifies that the old <b>fresults.dta</b> can be overwritten.
<p>
<b><u>Examples</u></b>
<p>
Take a dataset with 7 loci, the pairs of alleles at locus i are the variables
li_1 and li_2.
<p>
.swblock l1_1-l7_2, mv(mvdel)
<p>
<b>mvdel</b> was specified as the missing data mechanism and all subjects with any mis
&gt; sing data are deleted.
<p>
The following command changes the inclusion significance level to 1%
<p>
.swblock l1_1-l7_2, mv(mvdel) pvalue(0.01)
<p>
To store the results in a stata dataset do
<p>
.swblock l1_1-l7_2, mv(mvdel) pvalue(0.01) store replace
<p>
<b><u>Author</u></b>
<p>
Adrian Mander, Glaxo Smithkline, Harlow, UK.  Email adrian.p.mander@gsk.com
<p>
<b><u>Also see</u></b>
<p>
On-line:
Help for 
hapipf (MUST be installed),
ipf (MUST be installed)
hapblock (if installed). 
<p>
<p>
<p>
</pre>